dotnet/src/SemanticKernel/AI/OpenAI/Tokenizers/GPT3Tokenizer.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.SemanticKernel.AI.OpenAI.Tokenizers.Settings;
+
+namespace Microsoft.SemanticKernel.AI.OpenAI.Tokenizers;
+
+/// <summary>
+/// Port of GPT3 Javascript tokenizer recommended by OpenAI.
+/// See https://platform.openai.com/tokenizer and
+/// https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them
+/// </summary>
+public static class GPT3Tokenizer
+{
+    private static readonly ConcurrentDictionary<string, string> s_bpeCache = new();
+
+    private static ConcurrentDictionary<int, char>? s_bytesToUnicodeCache;
+
+    private static readonly Regex s_encodingRegex = new(
```

Comment:
[nit] Add a quick description on what this Regex is accomplishing.

Reply:
chatgpt said this: This regex is a pattern that can be used to split a string into different parts such as words, numbers, and punctuation marks. It matches different patterns in the string, including common contractions, words, numbers, and non-letter/non-digit characters. It can be used to tokenize text into different parts for further processing.

Reply:
I'll add a comment. It's used to handle English contractions